### PR TITLE
Remove unnecessary get_directory special cases for npm_and_yarn

### PR DESCRIPTION
**Pull request by ianlyons**
- The underlying DefinitelyTyped issue was resolved in https://github.com/microsoft/types-publisher/commit/910368a4f23336b0245d9abfe654ac577f36d90c
- The underlying Gatsby issue was resolved in https://github.com/gatsbyjs/gatsby/commit/d6d18e29c0d16d6b5f6030157381584dfb85d4a9

### CODE REVIEW

**Comment by ianlyons** [comment]
I didn't see any tests that exercised this case, but let me know if I missed something!

**Comment by jurre** [comment]
Thanks @ianlyons!

File: npm_and_yarn/lib/dependabot/npm_and_yarn/metadata_finder.rb [hunk 1]
```diff
@@ -123,18 +123,6 @@ def get_url(details)
 
       def get_directory(details)
         source_from_url = Source.from_url(get_url(details))
```
Comment (Nishnha) [inline npm_and_yarn/lib/dependabot/npm_and_yarn/metadata_finder.rb:125]: Looks like we can remove this line too
```suggestion
```

**Comment by ianlyons** [comment]
Ah yep, on it

**Comment by ianlyons** [comment]
@jurre are those bundler CI failures mine? happy to dig in

**Comment by ianlyons** [comment]
@jurre anything else I need to do to get this in?

**Comment by jurre** [comment]
> @jurre anything else I need to do to get this in?

No, apologies I just lost track of this one. I've rebased it and set up automerge for when CI passes

**Comment by yeikel** [comment]
@jurre a review is still needed for auto merge to work

**Review by landongrindheim — APPROVED** [review]

**Pull request merged**